File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.cpp
// Last updated: 18/10/2025, 09:01:09
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> ans;
        for(int i = 0; i < nums.size();i++){
            ans.push_back(nums[nums[i]]);
        }
        return ans;
    }
};

File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.java
// Last updated: 18/10/2025, 09:01:13
class Solution {
    public int[] buildArray(int[] nums) {
        int[] ans = new int[nums.length];
        for(int i = 0; i < nums.length; i++){
            ans[i] = nums[nums[i]];
        }

        return ans;
    }
}